zospy/analyses/mtf.py

```diff
@@ -237,3 +237,125 @@ def fft_through_focus_mtf_fromcfg(
+
+    For an in depth explanation of the parameters, see the Zemax OpticStudio user manual
+
+    Parameters
+    ----------
+    oss: zospy.core.OpticStudioSystem
+        A ZOSPy OpticStudioSystem instance. Should be sequential.
+    pupilsampling: str | int
+        The pupil sampling, either string (e.g. '64x64') or int.
+        The integer will be treated as a ZOSAPI Constants integer.
+    imagesampling: str | int
+        The image sampling, either string (e.g., '64x64') or int.
+        The integer will be treated as a ZOSAPI Constants integer.
+    imagedelta: float
+        The Image Delta, defaults to 0.0.
+    wavelength: str | int
+        The wavelength to use in the MTF. Either 'All' or an integer specifying the wavelength number.
+    field: str | int
+        The field to use in the MTF. Either 'All' or an integer specifying the field number.
+    mtftype: zospy.constants.Analysis.Settings.Mtf.MtfTypes.Modulation
+        The MTF type (e.g. `Modulation`) that is calculated.
+    maximumfrequency: float
+        The maximum frequency at which the MTF is calculated in lp/mm. Defaults to 150.0.
+    use_polarization: bool
+        Use polarization. Defaults to False.
+    use_dashes: bool
+        Use dashes. Defaults to False.
+    oncomplete: OnComplete | str
+        Defines behaviour upon completion of the analysis. Should be one of ['Close', 'Release', 'Sustain']. If 'Close',
+        the analysis will be closed after completion. If 'Release', the analysis will remain open in OpticStudio, but
+        the link with python will be destroyed. If 'Sustain' the analysis will be kept open in OpticStudio and the link
+        with python will be sustained. To enable interaction when oncomplete == 'Sustain', the OpticStudio Analysis
+        instance will be available in the returned AnalysisResult through AnalysisResult.Analysis. Defaults to 'Close'.
+
+    Returns
+    -------
+    AnalysisResult
+        A HuygensMTF analysis result
+    """
+    analysis_type = constants.Analysis.AnalysisIDM.HuygensMtf
+
+    # Create analysis
+    analysis = new_analysis(oss, analysis_type)
+
+    # Apply settings
+    analysis.Settings.PupilSampleSize = getattr(
+        constants.Analysis.SampleSizes, utils.zputils.standardize_sampling(pupilsampling)
+    )
+    analysis.Settings.ImageSampleSize = getattr(
+        constants.Analysis.SampleSizes, utils.zputils.standardize_sampling(imagesampling)
+    )
+    analysis.Settings.ImageDelta = imagedelta
+    analysis.set_wavelength(wavelength)
+    analysis.set_field(field)
+    analysis.Settings.Type = constants.process_constant(constants.Analysis.Settings.Mtf.HuygensMtfTypes, mtftype)
+    analysis.Settings.UsePolarization = use_polarization
+    analysis.Settings.UseDashes = use_dashes
+    analysis.Settings.MaximumFrequency = maximumfrequency
+
+    # Correct an API bug in setting API type for OpticStudio version <21.2
+    # TODO: not sure if this is needed here; commented for now
```

Comment:
This needs to be checked before we can merge this PR. @LucVV you are still running a version that's old enough, can you check this? Maybe we first need to extend the unit tests, because they do not yet change this parameter.
And @noahrbn which OpticStudio version do you use?

Reply:
The [post](https://community.zemax.com/zos-api-12/zos-api-setting-mtf-property-type-not-working-730) about this on the opticstudio forum only mentions `fft_through_focus_mtf`. However, I estimate that the fix might be required here also. I'll look into this

Reply:
I now see that `HuygensMtf` uses the enum `HuygensMtfTypes` rather than `MtfType`. In my version of OpticStudio, it is an enum that only contains `Modulation`. Thus, there is no way (and no need) to test this bugfix. However, I am in favor of keeping `mtftype` as parameter for future us.

Reply:
I checked this in OpticStudio 23 R2 and there the enum also has a single value. This means these lines can be removed.
I also prefer `mtftype` as parameter name over (variants of) `huygens_mtftype`.